R/plot_star_system.R

```diff
@@ -38,76 +40,100 @@
 #'                  habitable_zone = calculate_star_habitable_zone(data$st_lum[1]))
 #'
 #' @export
-plot_star_system = function(planet_data, spectral_type = NULL, habitable_zone = c(0, 0)) {
+plot_star_system = function(
+  planet_data,
+  spectral_type = " ",
+  habitable_zone = c(0, 0),
+  show_legend = FALSE
+) {
   assert_data_frame(planet_data)
   assert_numeric(habitable_zone, len = 2)
   assert_names(colnames(planet_data), must.include = c("pl_orbsmax", "pl_rade", "pl_dens"))
-  if (!is.null(spectral_type)) {
-    assert_character(spectral_type, len = 1, any.missing = FALSE)
-    assert_choice(spectral_type, choices = c("O", "B", "A", "F", "G", "K", "M"))
-  }
+  assert_character(spectral_type, len = 1, any.missing = FALSE)
+  assert_choice(spectral_type, choices = c(" ", "O", "B", "A", "F", "G", "K", "M"))
+
+  star_name = paste0(
+    "Central star",
+    if (!is.null(spectral_type)) paste0(", type ", spectral_type) else ""
+  )
 
-  # Create data frame for plotting the star system:
-  # First row is always the host star and following rows are the planets.
-  #    - randomize position on the orbit, with the star being in the center (0)
-  #    - rescale orbit for the visualization, keep the star at center (0). Add habitable zone edges.
-  #    - ensure star is the largest object with size 17
-  #    - map color based on spectral type or density
   plot_data = data.frame(
     orbit_offset = c(0, runif(nrow(planet_data), min = 0, max = 2 * pi), 0, 0),
     pl_orbsmax = c(0, .rescale_orbsmax(c(
       planet_data$pl_orbsmax,
-      min(habitable_zone), max(habitable_zone)
+      min(habitable_zone),
+      max(habitable_zone)
     ))),
-    pl_rade = c(17, planet_data$pl_rade, 0, 0), # Ensure star is the largest object
-    col = c(.map_star_color(spectral_type), .map_planet_color(planet_data$pl_dens), "", "")
+    pl_rade = c(17, planet_data$pl_rade, 0, 0),
+    type = c(spectral_type, .map_planet_type(planet_data$pl_dens), "HZ inner edge", "HZ outer edge")
   )
 
-  inner_hz = plot_data$pl_orbsmax[length(plot_data$pl_orbsmax) - 1]
-  outer_hz = plot_data$pl_orbsmax[length(plot_data$pl_orbsmax)]
-
+  inner_hz = plot_data$pl_orbsmax[nrow(plot_data) - 1]
+  outer_hz = plot_data$pl_orbsmax[nrow(plot_data)]
   plot_data = plot_data[1:(nrow(plot_data) - 2), ]
 
+  color_map = c(
+    " " = "white",
+    "M" = "red2",
+    "K" = "indianred2",
+    "G" = "yellow1",
+    "F" = "lightyellow",
+    "A" = "lavenderblush",
+    "B" = "lightsteelblue1",
+    "O" = "slateblue1",
+    "Gas-dominated planet" = "lightblue",
+    "Water/ice-rich planet" = "deepskyblue",
+    "Terrestial planet" = "darkorange3",
+    "Iron-rich planet" = "ivory3",
+    "Super-dense planet" = "gray20"
+  )
+
+  names(color_map)[1] = paste0("Central star, type: ", spectral_type)
```

Comment:
suggestion: The solution with dynamic naming of the color map is a bit awkward.

What I would do is to delete the _default_ white coloring and scrap the dynamic naming. This way:
- If star has no spectal type, it will be grey
- Grey star will not be included in the legend, as there is really no info about it

```suggestion
  color_map = c(
    "M" = "red2",
    "K" = "indianred2",
    "G" = "yellow1",
    "F" = "lightyellow",
    "A" = "lavenderblush",
    "B" = "lightsteelblue1",
    "O" = "slateblue1",
    "Gas-dominated planet" = "lightblue",
    "Water/ice-rich planet" = "deepskyblue",
    "Terrestial planet" = "darkorange3",
    "Iron-rich planet" = "ivory3",
    "Super-dense planet" = "gray20"
  )
```

but your solution ofc does the job if you want to keep it that is fine.